pyro/ops/jit.py

```diff
@@ -0,0 +1,80 @@
+import weakref
+import torch
+
+import pyro
+import pyro.poutine as poutine
+
+
+class compile(object):
+    """
+    Drop-in replacement for :func:`torch.jit.compile` that works with
+    Pyro functions that call :func:`pyro.param`.
+
+    The actual compilation artifact is stored in ``self.compiled``.
+    Call diagnostic methods on this attribute.
+
+    Example::
+
+        def model(x):
+            scale = pyro.param("scale", torch.tensor(0.5), constraint=constraints.positive)
+            return pyro.sample("y", dist.Normal(x, scale))
+
+        @pyro.ops.jit.compile(nderivs=1)
+        def model_log_prob_fn(x, y):
+            cond_model = pyro.condition(model, data={"y": y})
+            tr = pyro.poutine.trace(cond_model).get_trace(x)
+            return tr.log_prob_sum()
+    """
+    def __init__(self, **kwargs):
+        self._kwargs = kwargs
+        self.compiled = None
+        self._param_names = None
+
+    def __call__(self, fn):
+
+        assert self.compiled is None, \
+            "Cannot use a compile instance on more than one function"
+
+        def wrapper(*args, **kwargs):
```

Comment:
What do you think about adding a check like
```py
if self._fixed_kwargs is None:
    self._fixed_kwargs = kwargs
elif kwargs != self._fixed_kwargs:
    raise NotImplementedError('pyro.jit.compile does not support dynamic **kwargs')
```

Reply:
I don't know if that's true about `torch.jit.compile` though?

Reply:
ok